### PR TITLE
Refactoring of driver types; detection of existing drivers to guard against stomping data

### PR DESCRIPTION
**Pull request by iancmcc**

### CODE REVIEW

**Comment by iancmcc** [comment]
retest this please

File: volume/mocks/Driver.go [hunk 1]
```diff
@@ -1,22 +1,9 @@
-// Copyright 2015 The Serviced Authors.
```
Comment (smousa) [inline volume/mocks/Driver.go]: add back boilerplate

**Comment by iancmcc** [comment]
retest this please

**Comment by iancmcc** [comment]
retest this please

**Comment by iancmcc** [comment]
retest this please

**Comment by iancmcc** [comment]
retest this please

**Comment by iancmcc** [comment]
retest this please

**Comment by iancmcc** [comment]
retest this please

**Pull request merged**